oletools/olevba.py

```diff
@@ -811,6 +828,14 @@ def __init__(self, stream_path, variable, expected, value):
          ),
     'May run an executable file or a system command on a Mac (if combined with libc.dylib)':
         ('system', 'popen', r'exec[lv][ep]?'),
+    'May run an executable file or a system command using Excel 4 Macros (XLM/XLF)':
```

Comment:
I don't understand the change in lines 831-838, could you please explain why those regexes are needed, and why there are two keyword entries for EXEC and for REGISTER that look similar?

Reply:
This is to avoid the case when the output of oletools is:
```
|Suspicious|EXEC                |May run an executable file or a system       |
|          |                    |command using Excel 4 Macros (XLM/XLF)       |
|Suspicious|REGISTER            |May call a DLL using Excel 4 Macros (XLM/XLF)|
```
but, actually, there are no REGISTER or EXEC keywords called.

In recent XLM macro-based samples those keywords are extracted from this example code:
```
' 0006     40 FORMULA : Cell Formula - R45899C114 len=18 ptgNameV  *INCOMPLETE FORMULA PARSING* Could contain following functions: EXEC - Remaining, unparsed expression: bytearray(b'n\x00\x00\x00\x17\x06\x00pLBrGQ\x0eA\xac\x00')
' 0006     31 FORMULA : Cell Formula - R56495C91 len=9 ptgNameV  *INCOMPLETE FORMULA PARSING* Could contain following functions: REGISTER - Remaining, unparsed expression: bytearray(b'\x95\x00\x00\x00B\x01v\x80')
```
They are a false positive based on a bad parsing of the available output. Actually there is a `RUN` instead of those keywords. 

With this change, we avoid oletools to show those detections that can mislead analysts. Oletools will show only `REGISTER` and `EXEC` detection when they are not followed by that phrase, so when they are effectively extracted from the code. Otherwise it will explicilty say that the match is "Could contain following functions: EXEC" and not "EXEC" only. I hope to have been clear enough

Reply:
Do you have samples that I can check, which trigger this issue with EXEC/REGISTER? (I don't have VT, please upload here in a zip with password, or link to any.run, hybrid-analysis or similar)
Thanks!

Reply:
That behavior is caused by plugin_biff [here](https://github.com/DidierStevens/DidierStevensSuite/blob/300d7b79bb2a663ee741a76c286d7e2569157fed/plugin_biff.py#L1287).

This is a sample you can check with v0.55 and v0.56: https://app.any.run/tasks/921d3825-a2f7-4152-9b25-6dd9adb1cd49